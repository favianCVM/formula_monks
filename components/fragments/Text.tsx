import React from 'react';
import {Text as NativeText, TextProps} from 'react-native';
import {COLORS} from '../../styles/colors';

const Text = React.memo(
  ({children, style, testID = 'TextFragment'}: TextProps) => {
    return (
      <NativeText
        testID={testID}
        style={[
          {
            color: COLORS.text,
            fontSize: 16,
          },
          style,
        ]}>
        {children}
      </NativeText>
    );
  },
);

export {Text};
