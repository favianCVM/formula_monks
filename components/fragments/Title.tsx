import React from 'react';
import {Text, TextProps as NativeTextProps, StyleSheet} from 'react-native';
import {COLORS} from '../../styles/colors';
import {normalize} from '../../libs/normalizeSize';

interface TextProps extends NativeTextProps {}

const styles = StyleSheet.create({
  title: {
    fontSize: normalize(24),
    color: COLORS.text,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});

const Title = React.memo(({style, children}: TextProps) => {
  return <Text style={[styles.title, style]}>{children}</Text>;
});

export {Title};
