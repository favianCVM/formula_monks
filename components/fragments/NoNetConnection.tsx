import React from 'react';
import {Text} from './Text';
import Icon from 'react-native-vector-icons/FontAwesome6';
import {StyleSheet, View} from 'react-native';
import {COLORS} from '../../styles/colors';
import {normalize} from '../../libs/normalizeSize';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  center: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 10,
    backgroundColor: COLORS.red,
    gap: 14,
    alignSelf: 'flex-start',
    borderRadius: 10,
  },
  copy: {
    fontSize: normalize(24),
    fontWeight: 'bold',
    marginLeft: normalize(12),
  },
});

const NoNetConnection = React.memo(() => {
  return (
    <View testID="NoNetConnectionFragment" style={styles.container}>
      <View style={styles.center}>
        <Icon color={COLORS.gray} name="wifi" size={20} />
        <Text style={styles.copy}>no internet connection</Text>
      </View>
    </View>
  );
});

export {NoNetConnection};
