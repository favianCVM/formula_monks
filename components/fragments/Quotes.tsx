import React from 'react';
import {StyleSheet, TextProps} from 'react-native';
import {Text} from './Text';

const styles = StyleSheet.create({
  quotes: {
    fontWeight: 'bold',
    fontSize: 28,
  },
});

const Quotes = React.memo(({children, ...props}: TextProps) => (
  <Text {...props}>
    <Text style={styles.quotes}>“</Text>
    {children}
    <Text style={styles.quotes}>”</Text>
  </Text>
));

export {Quotes};
