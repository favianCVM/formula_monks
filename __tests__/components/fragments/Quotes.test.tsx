import React from 'react';
import {render} from '@testing-library/react-native';
import {Quotes} from '../../../components/fragments/Quotes';
import renderer from 'react-test-renderer';

describe('Quotes fragment', () => {
  it('renders text with the given value', () => {
    const {getByTestId} = render(<Quotes>hello world</Quotes>);

    const textFragment = getByTestId('QuoteFragment');
    expect(textFragment).toHaveTextContent('hello world');
  });

  it('renders no text', () => {
    const {getByTestId} = render(<Quotes></Quotes>);

    const textFragment = getByTestId('QuoteFragment');
    expect(textFragment).toHaveTextContent('“”');
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Quotes>hello world</Quotes>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
