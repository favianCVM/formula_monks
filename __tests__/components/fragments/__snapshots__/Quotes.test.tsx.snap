// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes fragment renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#1B1C1E",
        "fontSize": 16,
      },
      undefined,
    ]
  }
  testID="QuoteFragment"
>
  <Text
    style={
      [
        {
          "color": "#1B1C1E",
          "fontSize": 16,
        },
        {
          "fontSize": 28,
          "fontWeight": "bold",
        },
      ]
    }
    testID="TextFragment"
  >
    “
  </Text>
  hello world
  <Text
    style={
      [
        {
          "color": "#1B1C1E",
          "fontSize": 16,
        },
        {
          "fontSize": 28,
          "fontWeight": "bold",
        },
      ]
    }
    testID="TextFragment"
  >
    ”
  </Text>
</Text>
`;
